### ISSUE TITLE
Get item queries for items in shared folders fails with 404 error (API Regression)

### ISSUE THREAD

**Issue by abraunegg** [OneDrive/onedrive-api-docs#976]
#### Category
- [ ] Question
- [ ] Documentation issue
- [X] Bug

#### Expected or Desired Behavior

Previously (prior to potentially ~start of January) it was possible to query a drive item resource across LOCAL and SHARED folders.

As of at least 10th January, the query of SHARED folders / items is broken and returns a 404 - for example:

```
https://graph.microsoft.com/v1.0/me/drive/root:/./SHARED/:/?select=id... - SUCCESS
https://graph.microsoft.com/v1.0/users/{driveid}/drive/root:/./SHARED/ - SUCCESS
https://graph.microsoft.com/v1.0/me/drive/root:/./SHARED/1/:/?select=id... - FAILS
https://graph.microsoft.com/v1.0/users/{driveid}/drive/root:/./SHARED/1/ - FAILS
https://graph.microsoft.com/v1.0/me/drive/root:/./LOCAL/1/:/?select=id... - SUCCESS
https://graph.microsoft.com/v1.0/users/{driveid}/drive/root:/./LOCAL/1/ - SUCCESS
```

#### Observed Behavior
The query of items in a SHARED folder fails

#### Expected Behavior
The query of items in a SHARED folder should be successful - this previously worked

#### Steps to Reproduce
1. Create a shared folder with account 1 and share with account 2
2. Add shared folder to OneDrive under Files for account 2
3. Create an empty folder under the shared folder as account 2
4. Use MS Graph to query the sub folder object as account 2 - the top folder can be queried, however the sub folder or any item under the top folder returns the 404 where it used to return valid data.

**MS Graph Response of error**
```
{
    "error": {
        "code": "itemNotFound",
        "message": "Item does not exist",
        "innerError": {
            "request-id": "9ca3d393-0cf7-4141-b5fe-5de34068e9f1",
            "date": "2019-01-12T20:35:15"
        }
    }
}
```
**MS Graph response headers**
```
client-request-id: 9ca3d393-0cf7-4141-b5fe-5de34068e9f1
content-type: application/json
cache-control: private
request-id: 9ca3d393-0cf7-4141-b5fe-5de34068e9f1
```

**Comment by ificator** [comment]
Just to clarify, for both of the cases below you're making the call as `account 2`, and the `driveid` is for `account 2s` drive?

```
https://graph.microsoft.com/v1.0/me/drive/root:/./SHARED/1/:/?select=id... - FAILS
https://graph.microsoft.com/v1.0/users/{driveid}/drive/root:/./SHARED/1/ - FAILS
```

What does the response for the following look like? Do you see a `remoteItem` facet?

```
https://graph.microsoft.com/v1.0/me/drive/root:/./SHARED/:/?select=id... - SUCCESS
https://graph.microsoft.com/v1.0/users/{driveid}/drive/root:/./SHARED/ - SUCCESS
```

**Comment by abraunegg** [comment]
Response for `https://graph.microsoft.com/v1.0/users/66d53be8a5056eca/drive/root:/./SHARED/` as user 2:
```
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users('66d53be8a5056eca')/drive/root/$entity",
    "createdDateTime": "2019-01-14T18:57:11.827Z",
    "cTag": "adDo2NkQ1M0JFOEE1MDU2RUNBITc2NjMuNjM2ODMwODkxMjIwOTMwMDAw",
    "eTag": "aNjZENTNCRThBNTA1NkVDQSE3NjYzLjI",
    "id": "66D53BE8A5056ECA!7663",
    "lastModifiedDateTime": "2019-01-14T18:58:42.093Z",
    "name": "shared",
    "webUrl": "https://1drv.ms/u/s!AM-dUx_siH28igg",
    "createdBy": {
        "application": {
            "displayName": "OneDrive website",
            "id": "44048800"
        },
        "user": {
            "displayName": "",
            "id": "66d53be8a5056eca"
        }
    },
    "lastModifiedBy": {
        "user": {
            "displayName": "",
            "id": "bc7d88ec1f539dcf"
        }
    },
    "parentReference": {
        "driveId": "66d53be8a5056eca",
        "driveType": "personal",
        "id": "66D53BE8A5056ECA!101",
        "path": "/drive/root:"
    },
    "remoteItem": {
        "id": "BC7D88EC1F539DCF!1288",
        "size": 22029,
        "webUrl": "https://1drv.ms/u/s!AM-dUx_siH28igg",
        "fileSystemInfo": {
            "createdDateTime": "2019-01-14T18:54:43.2666667Z",
            "lastModifiedDateTime": "2019-01-14T18:54:43.2666667Z"
        },
        "folder": {
            "childCount": 2,
            "view": {
                "viewType": "thumbnails",
                "sortBy": "name",
                "sortOrder": "ascending"
            }
        },
        "parentReference": {
            "driveId": "bc7d88ec1f539dcf",
            "driveType": "personal"
        }
    }
}
```

**Comment by abraunegg** [comment]
If I add the next sub folder to shared `shared/1/` I get a 404 despite the 'shared' folder containing a sub folder & files:
```
https://graph.microsoft.com/v1.0/me/drive/root:/./SHARED/1/:/
https://graph.microsoft.com/v1.0/users/66d53be8a5056eca/drive/root:/./SHARED/1/
```

If I change to a 'local' (a user 2 item) folder:
```
https://graph.microsoft.com/v1.0/users/66d53be8a5056eca/drive/root:/./local/1/
https://graph.microsoft.com/v1.0/me/drive/root:/./local/1/:/
```
response
```
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users('mynas-test-acc%40outlook.com')/drive/root/$entity",
    "createdDateTime": "2019-01-14T19:10:57.297Z",
    "cTag": "adDo2NkQ1M0JFOEE1MDU2RUNBITc2NjcuNjM2ODMwODk4ODM0NTMwMDAw",
    "eTag": "aNjZENTNCRThBNTA1NkVDQSE3NjY3LjA",
    "id": "66D53BE8A5056ECA!7667",
    "lastModifiedDateTime": "2019-01-14T19:11:23.453Z",
    "name": "1",
    "size": 11017,
    "webUrl": "https://1drv.ms/f/s!AMpuBaXoO9Vmu3M",
    "createdBy": {
        "application": {
            "displayName": "OneDrive website",
            "id": "44048800"
        },
        "user": {
            "displayName": "",
            "id": "66d53be8a5056eca"
        }
    },
    "lastModifiedBy": {
        "application": {
            "displayName": "OneDrive website",
            "id": "44048800"
        },
        "user": {
            "displayName": "",
            "id": "66d53be8a5056eca"
        }
    },
    "parentReference": {
        "driveId": "66d53be8a5056eca",
        "driveType": "personal",
        "id": "66D53BE8A5056ECA!7666",
        "name": "local",
        "path": "/drive/root:/local"
    },
    "fileSystemInfo": {
        "createdDateTime": "2019-01-14T19:10:57.296Z",
        "lastModifiedDateTime": "2019-01-14T19:10:57.296Z"
    },
    "folder": {
        "childCount": 0,
        "view": {
            "viewType": "thumbnails",
            "sortBy": "name",
            "sortOrder": "ascending"
        }
    }
}
```

**Comment by ificator** [comment]
Thanks for you patience on this one, I wanted to try a few things before I responded.

The short answer is: I'm not sure how this scenario could ever have worked.

When a folder in `User B`'s drive is "added" to `User A`'s drive (e.g. using "Add to my drive" on the web) `User A` will only have a "pointer" persisted in their drive. When an app queries that "pointer" they'll get a `remoteItem` facet back that contains information about how to query the folder in `User B`'s drive. There isn't any process that syncs any information about the descendants of the folder in `User B`'s drive into the `User A`'s drive, and so making a request against `/drives/{usera}/root:/SHARED/1` would only ever return a 404.

**Comment by abraunegg** [comment]
@ificator 
OK .. I think youve got the situation backwards:

1. User A creates a folder / path of 'whatever' and shares that with User B
2. User B adds this folder to their OneDrive under 'Files' via web interface
3. User B can see files / folders, create files and folders under OneDrive web interface
4. User B uses MS Graph / API to query same files / folders / path tree on OneDrive
5. User B can query the 'top level folders' on OneDrive via MS Graph
6. User B can query any 'sub folder' on MS Graph for folders that are not a 'shared folder'
7. User B cannot query any 'sub folder' on MS Graph for folders that originate in a shared folder

This used to work and is a recent API regression - you used to be able to query each item in 'your' OneDrive account as the whole point of querying a 'path' in your OneDrive folder is to see whats there - you can see it via OneDrive web UI - why cant you now query it via API ...

The thing is - as User B - if I use `https://graph.microsoft.com/v1.0/me/drive/recent` after modifying a file / folder which is on the 'shared' drive - this returns information.

**Comment by abraunegg** [comment]
@ificator 
Any update on looking at this issue?

Happy to setup a WebEx / conf call / show what I am seeing if this helps to get some clarity into this issue.

**Comment by abraunegg** [comment]
@ificator
Any update on looking at this issue?

Happy to setup a WebEx / conf call / show what I am seeing if this helps to get some clarity into this issue.

**Comment by Alexduartecwb** [comment]
I apologize if I'm making any mistake... but it was really working fine here until last monday, when the kernel was updated.

I'm using only shared folders.
I cannot create/upload any new folder/file. 
Upload of modified files, ok.
Download is ok.

This is my first comment her, so, if I'm doing something wrong just let me know.

**Comment by abraunegg** [comment]
@ificator
Any update on looking at this issue?

Happy to setup a WebEx / conf call / show what I am seeing if this helps to get some clarity into this issue.

**Comment by 0x3333** [comment]
No updated on this issue?

**Comment by frispete** [comment]
Obviously not. 

Since MS is new on the Open Source Stage, they're currently in the handwaving train phase with this issue...

**Comment by wiryonolau** [comment]
Just wondering, this API is not exclusive right ? so It should be the same for onedrive  in android / ios / mac if any... or they use another exclusive secret API for those client ? T_T

**Comment by abraunegg** [comment]
@ificator 
As of 6am 3rd March 2019 - this is now again working - able to query objects.

Any idea what the change was?

**Comment by abraunegg** [comment]
@ificator 
For future reference - validation this is working:
```
Uploading new items of .
OneDrive Client requested to create remote path: ./shared/newdir
The requested directory to create was not found on OneDrive - creating remote directory: ./shared/newdir
Successfully created the remote directory ./shared/newdir on OneDrive
Uploading file ./shared/newdir/asdf.txt ... done.
Remaining free space: 5368709115
Applying changes of Path ID: 66D53BE8A5056ECA!101
Applying changes of Path ID: BC7D88EC1F539DCF!1288
```
![shared-upload](https://user-images.githubusercontent.com/4956234/53982469-1dde7300-4169-11e9-962f-11fbd6d11fad.jpg)
![shared-upload-sucess](https://user-images.githubusercontent.com/4956234/53982475-25058100-4169-11e9-92e0-9c18a9c3c70c.jpg)

Previously, the query to determine if 'newdir' already existed was failing, but prior to 10th January this worked - so now we are back to a working state for query of shared object

**Comment by kguidonimartins** [comment]
Hello there! Thanks for this onedrive implementation! 

I have been working with git inside a onedrive folder. I'm syncing some medium-size CSV file and versioning the code used for analysis. 

My first issue is related to duplication of ZIP files. When I sync ondedrive, vostro (my pc name on linux) is appended to ZIP filename. This is true for all of my ZIP files.

My second issue is related to .git folder and contents inside it. Onedrive seems don't sync the hooks scripts and others contents (see log below).

```
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Uploading file spp-ameacadas/.git/index ... done.
Uploading file ./spp-ameacadas/data/raw/GBIF/0037587-181108115102211-vostro.zip ...
Uploading 100% |oooooooooooooooooooooooooooooooooooooooo| DONE IN 00:00:45                                                                             
 done.
Uploading file ./spp-ameacadas/data/raw/iDigBio/aafd7529-2d4f-4cd6-81c4-076f0c880d29-vostro-2.zip ...
Uploading 100% |oooooooooooooooooooooooooooooooooooooooo| DONE IN 00:01:45                                                                             
 done.
Uploading file ./spp-ameacadas/data/raw/iDigBio/aafd7529-2d4f-4cd6-81c4-076f0c880d29-vostro.zip ...
Uploading 100% |oooooooooooooooooooooooooooooooooooooooo| DONE IN 00:03:19                                                                             
 done.

ERROR: Unable to query the following path due to OneDrive API regression: ./spp-ameacadas/.git/hooks
ERROR: Refer to https://github.com/OneDrive/onedrive-api-docs/issues/976 for further details
WORKAROUND: Manually create the path above on OneDrive to workaround API issue

Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/prepare-commit-msg.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/pre-applypatch.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/update.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/pre-commit.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/pre-push.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/applypatch-msg.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/post-update.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/commit-msg.sample
Skipping uploading this new file as parent path is not in the database: ./spp-ameacadas/.git/hooks/pre-rebase.sample

ERROR: Unable to query the following path due to OneDrive API regression: ./spp-ameacadas/.git/objects
ERROR: Refer to https://github.com/OneDrive/onedrive-api-docs/issues/976 for further details
WORKAROUND: Manually create the path above on OneDrive to workaround API issue

```

**Comment by abraunegg** [comment]
@kguidonimartins
This issue is actually resolved, and fixed in v2.2.6 of the client. Suggest you rebuild your client from https://github.com/abraunegg/onedrive and if you have any further issue, raise a case on my GitHub branch.
